packages/airnode-validator/src/ois/ois.ts

```diff
@@ -106,11 +113,50 @@ export const operationSchema = z.object({
   parameters: z.array(operationParameterSchema),
 });
 
-export const pathSchema = z.record(operationSchema);
+export const httpStatusCodes = z.union([z.literal('get'), z.literal('post')]);
+
+export const pathSchema = z.record(httpStatusCodes, operationSchema);
+
+const ensurePathParametersExist: ValidatorRefinement<SchemaType<typeof pathsSchema>> = (paths, ctx) => {
+  forEach(paths, (pathData, rawPath) => {
+    forEach(pathData, (paramData, httpMethod) => {
+      const parameters = paramData!.parameters;
+      // Match on anything in the path that is braces
+      // i.e. The path /users/{id}/{action} will match ['{id}', '{action}']
+      const regex = /\{([^}]+)\}/g;
+      const matches = rawPath.match(regex)?.map(removeBraces) ?? [];
+
+      // Check that all path parameters are defined
+      matches.forEach((match) => {
+        const parameter = parameters.find((p) => p.in === 'path' && p.name === match);
+        if (!parameter) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Path parameter "${match}" is not found in "parameters"`,
+            path: [rawPath, httpMethod, 'parameters'],
+          });
+        }
+      }, rawPath);
+
+      // Check that all parameters are used
+      parameters.forEach((p, index) => {
+        if (p.in === 'path' && !matches.includes(p.name)) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `Parameter "${p.name}" is not used in the URL path`,
```

Comment:
"is not used in" sounds a bit weird to me. Maybe something like:

```suggestion
            message: `Parameter "${p.name}" was not found in the URL path`,
```

Reply:
"is not found" sounds good to me.